Failing to delete temporary file in FRMS6 test on Win 11

Looks like a reference lives on, despite the `del`. Reporting as an Issue since it leaves an 8.6 GB file behind.

```
=================================== ERRORS ====================================
___________ ERROR at teardown of test_compare_backends_sparse[True] ___________

tmpdir_factory = TempdirFactory(_tmppath_factory=TempPathFactory(_given_basetemp=None, _trace=<pluggy._tracing.TagTracerSub object at 0...owsPath('C:/Users/weber/AppData/Local/Temp/pytest-of-unknown/pytest-15'), _retention_count=3, _retention_policy='all'))

    @pytest.fixture(scope='module')
    def default_frms6_raw(tmpdir_factory):
        fn = tmpdir_factory.mktemp("data").join("frms6.raw")
        # we use a memory mapped file to make this work
        # on machines that can't hold the full dataset in memory
        data = np.memmap(str(fn), mode='w+', shape=(256, 256, 264, 264), dtype='uint16')
        view = reshaped_view(data, (256*256, 264, 264))
        root, ext = os.path.splitext(FRMS6_TESTDATA_PATH)
        files = list(sorted(glob.glob(root + '*.frms6')))
        blocksize = 15
        offset = 0
        # we skip the first file, it contains a zero reference
        for f in files[1:]:
            raw_shape = stemtool.util.pnccd.Frms6Reader.getDataShape(f)
            frame_count = raw_shape[-1]
            # We go blockwise to reduce memory consumption
            for start in range(0, frame_count, blocksize):
                stop = min(start+blocksize, frame_count)
                block = _read_block(f, raw_shape, start, stop)
                view[offset + start:offset + stop] = _unfold_block(block)
            offset += frame_count
    
        yield data
    
        # Might be needed for windows, so we don't keep any views, mmaps or open
        # file handles around:
        del view
        del data
>       os.unlink(fn)
E       PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\weber\\AppData\\Local\\Temp\\pytest-of-unknown\\pytest-15\\data5\\frms6.raw'

tests\io\datasets\test_frms6.py:126: PermissionError
```

--- a/docs/source/releasing.rst
+++ b/docs/source/releasing.rst
@@ -60,6 +60,7 @@ When planning a release, create a new issue with the following checklist:
         * [ ] LiberTEM-holo
         * [ ] LiberTEM-blobfinder
         * [ ] ptychography40
+        * [ ] LiberTEM-iCoM
     * [ ] Handle deprecation, search the code base for `DeprecationWarning`
           that are supposed to be removed in that release.
     * [ ] GUI dependency update with `npm install`

--- a/setup.py
+++ b/setup.py
@@ -133,7 +133,9 @@ setup(
         "sparse",
         "distributed>=2.19.0",
         "dask!=2023.6.1",  # work around broken ipython exception handler
-        "click",
+        # dask-scheduler is incompatible with newer click versions
+        "click<8.1;( python_version<'3.8')",
+        "click;( python_version>='3.8')",
         "tornado>=5",
         "matplotlib",
         "pillow",

--- a/tests/io/datasets/test_frms6.py
+++ b/tests/io/datasets/test_frms6.py
@@ -122,6 +122,7 @@ def default_frms6_raw(tmpdir_factory):
     # Might be needed for windows, so we don't keep any views, mmaps or open
     # file handles around:
     del view
+    data._mmap.close()
     del data
     os.unlink(fn)
 

--- a/tests/server/test_cancel.py
+++ b/tests/server/test_cancel.py
@@ -82,7 +82,8 @@ async def test_cancel_udf_job(
         num_seen = 0
         while not done:
             msg = json.loads(await ws.recv())
-            num_seen += 1
+            if msg['messageType'] != "JOB_PROGRESS":
+                num_seen += 1
             if msg['messageType'] == 'TASK_RESULT':
                 assert_msg(msg, 'TASK_RESULT')
                 assert msg['job'] == job_uuid

